Index: public_html/sites/all/modules/contrib/fieldable_panels_panes/includes/PanelsPaneController.class.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- public_html/sites/all/modules/contrib/fieldable_panels_panes/includes/PanelsPaneController.class.php	(revision 3bfa99ebd6d9059554713405ec1a8d2c6fb65dd7)
+++ public_html/sites/all/modules/contrib/fieldable_panels_panes/includes/PanelsPaneController.class.php	(revision )
@@ -326,7 +326,7 @@
   public function create($values) {
     $entity = (object) array(
       'bundle' => $values['bundle'],
-      'language' => LANGUAGE_NONE,
+      'language' => 'en',  // HACK: Temporarily create all FPP entities with EN as language.
       'is_new' => TRUE,
     );
 
