Index: sites/all/modules/contrib/filefield_paths/filefield_paths.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- sites/all/modules/contrib/filefield_paths/filefield_paths.module	(revision dc87e2718dfa815c8992adc82bb134530a9f9cf4)
+++ sites/all/modules/contrib/filefield_paths/filefield_paths.module	(revision )
@@ -444,8 +444,11 @@
   // Build regular expression pattern.
   $pattern = "/({$prefixes})({$styles})*({$paths}){$itok}/";
 
+  $new_escaped = addslashes($new);
+  $old_escaped = addslashes($old);
+
   // Create an anonymous function for the replacement via preg_replace_callback.
-  $callback = create_function('$matches', "return filefield_paths_replace_path_callback(\$matches, '{$new}', '{$old}');");
+  $callback = create_function('$matches', "return filefield_paths_replace_path_callback(\$matches, '{$new_escaped}', '{$old_escaped}');");
 
   $fields = field_info_fields();
   foreach ($fields as $name => $field) {
