Index: public_html/sites/all/modules/contrib/ds/ds.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- public_html/sites/all/modules/contrib/ds/ds.module	(revision 7bbbad4294a0826b6f2bb46d899ab77902f7ee00)
+++ public_html/sites/all/modules/contrib/ds/ds.module	(revision )
@@ -692,11 +692,11 @@
 
       // Add extras classes to the region.
       if (empty($layout['flexible'])) {
-        $vars[$region_name . '_classes'] = !empty($layout['settings']['classes'][$region_name]) ? ' ' . implode(' ', $layout['settings']['classes'][$region_name]) : '';
+        $vars[$region_name . '_classes'] .= !empty($layout['settings']['classes'][$region_name]) ? ' ' . implode(' ', $layout['settings']['classes'][$region_name]) : '';
 
         // Token support for region classes.
         if (module_exists('token') && isset($vars[$object])) {
-          $vars[$region_name . '_classes'] = token_replace($vars[$region_name . '_classes'], array($object => $vars[$object]), array('clear' => TRUE, 'sanitize' => TRUE));
+          $vars[$region_name . '_classes'] .= token_replace($vars[$region_name . '_classes'], array($object => $vars[$object]), array('clear' => TRUE, 'sanitize' => TRUE));
         }
       }
       // Add a wrapper to the region.
