Index: public_html/sites/all/modules/contrib/ds/ds.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- public_html/sites/all/modules/contrib/ds/ds.module	(revision 18cb4438a7b08af18fd1990e2e28f1a446cba727)
+++ public_html/sites/all/modules/contrib/ds/ds.module	(revision )
@@ -704,7 +704,7 @@
         }
       }
       // Add a wrapper to the region.
-      if (empty($layout['flexible'])) {
+      if (empty($layout['flexible']) && !isset($vars[$region_name . '_wrapper'])) {
         $vars[$region_name . '_wrapper'] = isset($layout['settings']['wrappers'][$region_name]) ? $layout['settings']['wrappers'][$region_name] : 'div';
       }
     }
@@ -713,7 +713,9 @@
     $vars['rendered_by_ds'] = TRUE;
 
     // Add a layout wrapper.
+    if (!isset($vars['layout_wrapper'])) {
-    $vars['layout_wrapper'] = isset($layout['settings']['layout_wrapper']) ? $layout['settings']['layout_wrapper'] : 'div';
+      $vars['layout_wrapper'] = isset($layout['settings']['layout_wrapper']) ? $layout['settings']['layout_wrapper'] : 'div';
+    }
 
     // Add layout attributes if any.
     if (!isset($vars['layout_attributes'])) {
