diff --git a/public_html/sites/all/modules/contrib/adminimal_admin_menu/js/slicknav/jquery-no-conflict.slicknav.js b/public_html/sites/all/modules/contrib/adminimal_admin_menu/js/slicknav/jquery-no-conflict.slicknav.js
index ff7af8b..26b702f 100644
--- a/public_html/sites/all/modules/contrib/adminimal_admin_menu/js/slicknav/jquery-no-conflict.slicknav.js
+++ b/public_html/sites/all/modules/contrib/adminimal_admin_menu/js/slicknav/jquery-no-conflict.slicknav.js
@@ -378,7 +378,7 @@
 Drupal.admin.behaviors.responsivemenu = function (context, settings, $adminMenu) {
 
     $('#admin-menu-menu-responsive').slicknav({
-		label: 'Menu',
+		label: 'Admin Menu',
 		prependTo:'body',
 		closedSymbol: "<i class=\"closed\"></i>",
 		openedSymbol: "<i class=\"open\"></i>",
@@ -429,4 +429,4 @@ Drupal.admin.behaviors.responsiveshortcuts = function (context, settings, $admin
   $('#admin-menu-shortcuts-responsive #edit-shortcuts').remove();
 
 };
-})($jQueryAdminimal);
\ No newline at end of file
+})($jQueryAdminimal);
